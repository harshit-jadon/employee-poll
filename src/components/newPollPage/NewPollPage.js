import "./NewPollPage.css";
import { useState } from "react";
import { useNavigate } from "react-router-dom";
import { connect } from "react-redux";
import { handleAddQuestion } from "../../actions/questions";

const NewPollPage = ({ handleAddQuestion }) => {
  
  const [optOne, setOptOne] = useState("");
  const [optTwo, setOptTwo] = useState("");

  const navigate = useNavigate();

  const submitNewPoll = (e) => {
    handleAddQuestion(optOne, optTwo);
    navigate("/");
    e.preventDefault();
  };

  return (
    <div className="poll-div">
      <h3>Would You Rather</h3>
      <h6>Create Your Own Poll</h6>
      <form className="form-div" onSubmit={submitNewPoll}>
        <div className="label-input-dev">
          <label
            className="label"
            htmlFor="firstOption"
            data-testid="firstOptionLabel"
          >
            First Option
          </label>
          <div>
            <input
              type="text"
              id="firstOption"
              data-testid="firstOption"
              value={optOne}
              onChange={(e) => setOptOne(e.target.value)}
              className="input"
              placeholder="Option One"
            />
          </div>
        </div>

        <div className="label-input-dev">
          <label htmlFor="secondOption" data-testid="secondOptionLabel">
            Second Option
          </label>
          <div>
            <input
              type="text"
              id="secondOption"
              data-testid="secondOption"
              value={optTwo}
              onChange={(e) => setOptTwo(e.target.value)}
              className="input"
              placeholder="Option Two"
            />
          </div>
        </div>

        <div className="submit-btn-poll">
          <button
            className="submit-poll"
            type="submit"
            data-testid="submit-poll"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

const mapDispatchToProps = {
  handleAddQuestion,
};

export default connect(null, mapDispatchToProps)(NewPollPage);
