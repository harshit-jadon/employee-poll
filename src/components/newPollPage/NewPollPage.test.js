import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import NewPollPage from "./NewPollPage";
import { store } from "../../store";
import { BrowserRouter } from "react-router-dom";

describe("NewPoll", () => {
  it("Render the New PollPage component", () => {
    const NewPollPagecomponent = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );
    expect(NewPollPagecomponent).toBeDefined();
    expect(NewPollPagecomponent).toMatchSnapshot();
  });

  it('should render the form inputs in th enewpoll page component', () => {
    const NewPollPagecomponent = render(
        <Provider store={store}>
          <BrowserRouter>
            <NewPollPage />
          </BrowserRouter>
        </Provider>
      );
  expect(NewPollPagecomponent.getByTestId('label-first-option')).toBeInTheDocument();
  expect(NewPollPagecomponent.getByTestId('input-first-option')).toBeInTheDocument();
  expect(NewPollPagecomponent.getByTestId('label-second-option')).toBeInTheDocument();
  expect(NewPollPagecomponent.getByTestId('input-second-option')).toBeInTheDocument();
  expect(NewPollPagecomponent.getByTestId('submit-button')).toBeInTheDocument();
});

  it("should display all elements in the newpoll component", () => {
    const NewPollPagecomponent = render(
      <Provider store={store}>
        <BrowserRouter>
          <NewPollPage />
        </BrowserRouter>
      </Provider>
    );

    const firstOptionLabel =
      NewPollPagecomponent.getByTestId("label-first-option");
    const firstOptionInput =
      NewPollPagecomponent.getByTestId("input-first-option");
    expect(firstOptionLabel.textContent).toBe("First Option");
    fireEvent.change(firstOptionInput, { target: { value: "Javascript" } });
    expect(firstOptionInput.value).toBe("Javascript");

    const secondOptionLabel = NewPollPagecomponent.getByTestId(
      "label-second-option"
    );
    const secondOptionInput = NewPollPagecomponent.getByTestId(
      "input-second-option"
    );
    expect(secondOptionLabel.textContent).toBe("Second Option");
    fireEvent.change(secondOptionInput, {
      target: { value: "Web- development" },
    });
    expect(secondOptionInput.value).toBe("Web- development");

    const submitButtonElement =
      NewPollPagecomponent.getByTestId("submit-button");
    expect(submitButtonElement.textContent).toBe("Submit");
  });


  
});
