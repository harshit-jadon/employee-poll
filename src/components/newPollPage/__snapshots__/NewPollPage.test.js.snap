// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll Render the New PollPage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-div"
      >
        <h3
          data-testid="poll-div-comp"
        >
          Would You Rather
        </h3>
        <h6>
          Create Your Own Poll
        </h6>
        <form
          class="form-div"
        >
          <div
            class="label-input-dev"
          >
            <label
              class="label"
              data-testid="label-first-option"
              for="firstOption"
            >
              First Option
            </label>
            <div>
              <input
                class="input"
                data-testid="input-first-option"
                id="firstOption"
                placeholder="Option One"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="label-input-dev"
          >
            <label
              data-testid="label-second-option"
              for="secondOption"
            >
              Second Option
            </label>
            <div>
              <input
                class="input"
                data-testid="input-second-option"
                id="secondOption"
                placeholder="Option Two"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="submit-btn-poll"
          >
            <button
              class="disable-submit"
              data-testid="submit-button"
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-div"
    >
      <h3
        data-testid="poll-div-comp"
      >
        Would You Rather
      </h3>
      <h6>
        Create Your Own Poll
      </h6>
      <form
        class="form-div"
      >
        <div
          class="label-input-dev"
        >
          <label
            class="label"
            data-testid="label-first-option"
            for="firstOption"
          >
            First Option
          </label>
          <div>
            <input
              class="input"
              data-testid="input-first-option"
              id="firstOption"
              placeholder="Option One"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="label-input-dev"
        >
          <label
            data-testid="label-second-option"
            for="secondOption"
          >
            Second Option
          </label>
          <div>
            <input
              class="input"
              data-testid="input-second-option"
              id="secondOption"
              placeholder="Option Two"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="submit-btn-poll"
        >
          <button
            class="disable-submit"
            data-testid="submit-button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
